packages/oauth2/src/common/algorithm/algorithm-transform.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * Algorithm transformation utilities for JWA and COSE
+ *
+ * This module provides utilities to transform between JWA (JSON Web Algorithms)
+ * signature algorithm identifiers and fully-specified COSE (CBOR Object Signing and Encryption)
+ * algorithm identifiers.
+ *
+ * Based on RFC 9864: Fully-Specified Algorithms for JOSE and COSE
+ * https://www.rfc-editor.org/rfc/rfc9864.html
+ */
+
+import { Oauth2Error } from '../../error/Oauth2Error'
+
+/**
+ * JWA (JSON Web Algorithms) signature algorithm identifiers
+ *
+ * From RFC 7518 (JWA) and RFC 9864 (Fully-Specified Algorithms)
+ */
+enum JwaSignatureAlgorithm {
+  // EdDSA algorithms - RFC 9864 Section 2.2
+  Ed25519 = 'Ed25519',
+  Ed448 = 'Ed448',
+
+  // Deprecated polymorphic EdDSA - RFC 9864 Section 4.1.2
+  // Maps to Ed25519 as it's the most common use case (similar to WebAuthn's approach)
+  EdDSA = 'EdDSA',
+
+  // ECDSA algorithms - RFC 9864 Section 2.1
+  // JWA ECDSA algorithms are already fully-specified
+  ES256 = 'ES256',
+  ES384 = 'ES384',
+  ES512 = 'ES512',
+  ES256K = 'ES256K',
+
+  // RSA algorithms - RFC 7518
+  RS256 = 'RS256',
+  RS384 = 'RS384',
+  RS512 = 'RS512',
+  PS256 = 'PS256',
+  PS384 = 'PS384',
+  PS512 = 'PS512',
+}
+
+/**
+ * Mapping of JWA signature algorithm identifiers to fully-specified COSE algorithm identifiers
+ *
+ * From RFC 9864:
+ * - EdDSA algorithms (Section 2.2)
+ * - ECDSA algorithms (Section 2.1) - JWA ECDSA algorithms are already fully-specified
+ *
+ * Note: JWA ECDSA algorithms (ES256, ES384, ES512) are already fully-specified,
+ * while COSE ECDSA algorithms with the same names are polymorphic and deprecated.
+ * The fully-specified COSE equivalents use different names (ESP256, ESP384, ESP512).
+ */
+const JWA_SIGNATURE_TO_COSE_ALGORITHM_MAP = {
+  // EdDSA algorithms - RFC 9864 Section 2.2
```

Comment:
Maybe these comments can be jsdoc? Would be useful info for the user I think.

Reply:
There' already references to the RFC and sections in the jsdoc, so not sure what we need to add? I made an additional commit to reference the RFC in more places